server/engagement/views.py

```diff
@@ -1,51 +1,146 @@
+                    {'error': f'{field} is required'},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
 
-    @transaction.atomic
-    def post(self, request, *args, **kwargs):
-        discord_id = request.data.get("discord_id")
-        channel_id = request.data.get("channel_id")
         try:
-            user = get_object_or_404(User, discord_id=discord_id)
-
-            (
-                stats,
-                created,
-            ) = DiscordMessageStats.objects.select_for_update().get_or_create(
-                member_id=user.id, channel_id=channel_id, defaults={"message_count": 1}
+            expires = request.data['expires']
+            try:
+                expires = parse_date(expires)
+                expires = expires.astimezone(timezone.utc)
+            except ValueError:
+                return Response(
+                    {"detail": "Invalid time format. Please use ISO format (with Z)."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                    )
+            
+            session = AttendanceSession.objects.create(
+                title=request.data['title'],
+                key=request.data['key'],
+                expires=expires
             )
 
-            if not created:
-                stats.message_count = F("message_count") + 1
-                stats.save()
-                stats.refresh_from_db()
+            return Response({
+                'session': {
+                    'session_id': session.session_id,
+                    'title': session.title,
+                    'key': session.key,
+                    'expires': session.expires
+                }
+            }, status=status.HTTP_201_CREATED)
 
-            logger.info(
-                "member %s has %d messages in channel %s",
-                user.id,
-                stats.message_count,
-                channel_id,
+        except ValueError as e:
+            return Response(
+                {'error': 'Invalid date or timestamp format'},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        except Exception as e:
+            return Response(
+                {'error': str(e)},
+                status=status.HTTP_400_BAD_REQUEST
             )
 
-            return Response(status=status.HTTP_202_ACCEPTED)
+class GetAttendanceSessions(generics.ListAPIView):
+    permission_classes = [IsAdmin|IsApiKey]
+    serializer_class = AttendanceSessionSerializer
+    
+    def get_queryset(self):
+        return AttendanceSession.objects.all()
 
-        except Http404:
-            logger.error("member not found for discord_id: %s", discord_id)
+class GetMemberAttendanceSessions(generics.ListAPIView):
+    permission_classes = [IsAdmin|IsApiKey]
+    serializer_class = AttendanceSessionSerializer
+    
+    def get_queryset(self):
+        user = get_object_or_404(User, id=self.kwargs['id'])
+        return AttendanceSession.objects.filter(attendees=user) 
```

Comment:
Would filtering like this work in general? Since attendees is a list (I think), shouldn't we be including `AttendanceSession`s that contain the user in its attendee list, not when the attendee list is equal to the user? Could you add some tests that validate this functionality with multiple users?

Reply:
I have no idea why I wrote this

Reply:
actually, looks like this works fine